mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -172,10 +179,10 @@ static void insertCopyFromOtherArg(PatternRewriter &b, Location loc,
                                    Value dest) {
   LLVM_DEBUG(llvm::dbgs() << "Src type: " << srcOp.getType()
                           << " dest type: " << op.dest().getType() << "\n");
-  assert(srcOp.getType().cast<ShapedType>().getShape() ==
-             op.dest().getType().cast<ShapedType>().getShape() &&
-         "shape of extra fusion arguments matches shape of C tensor");
-
+  /*  assert(srcOp.getType().cast<ShapedType>().getShape() ==
```

Comment:
This assertion is, IIRC, a key invariant in the code below - it's there for good reason. If you're removing this assertionn (which should be removed, not comment out) please explain how either
1. The asserted invariant did not need to be true; or
2. You have modified the code below to relax the invariant

One note I have is that, in the code as I remember writing it, the fix for the extra argument's shape not matching the output tensor's shape was to `TransformOp` said extra argument tensor until it's shape matched. Why can this approach not be used for this new bug?

Reply:
I want 2, because this change is trying to support the case when the extra arg's shape is different to the output shape.
Instead, can we check if they're compatible?